py/farm_ng/core/event_service_recorder.py

```diff
@@ -144,6 +165,14 @@ async def subscribe(
         event: event_pb2.Event
         payload: bytes
         async for event, payload in client.subscribe(subscription, decode=False):
+            if event.uri in header_uris:
+                # Handle header messages - typically these will be metadata or calibrations
+                # This assumes they are only published once (likely with latching=True),
+                # as we break the subscription after the first header message is received
+                self.logger.info("header received: %s", event)
+                self.header_deque.append((event, payload))
```

Comment:
This seems risky. Can you replace a header message if the uri is the same?

Reply:
Now you cannot. See other conversation thread.